### ISSUE TITLE
Unable to create directory /Users/runner/Library/Caches/pylint

### ISSUE THREAD

**Issue by ajeetdsouza** [PyCQA/pylint#5347]
### Bug description

I am using pylint in some tests in my code. When executing the tests in parallel, they sometimes error out like this:

```shell
$ echo "<generated python code>" | pylint --from-stdin foo # there are 24 tests running this command in parallel

Unable to create directory /Users/runner/Library/Caches/pylint
```

I'm running the same tests on Linux, but haven't seen this problem occur there (yet).

### Command used

```shell
`echo "<generated python code>" | pylint --from-stdin foo`
```

### Pylint output

```shell
`Unable to create directory /Users/runner/Library/Caches/pylint`
```

### Expected behavior

pylint should create cache directories in a thread-safe manner.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.9.6 (default, Jun 28 2021, 08:57:49)
[GCC 10.3.0]
```

### OS / Environment

macOS 11

**Comment by Pierre-Sassoulas** [comment]
Does it works if you're launching only one test ?

**Comment by ajeetdsouza** [comment]
Yes. Most of the time, all 24 tests run successfully, but sometimes 1 of them might fail.
